viewer/packages/360-images/src/collection/DefaultImage360Collection.ts

```diff
@@ -182,4 +205,63 @@ export class DefaultImage360Collection implements Image360Collection {
       entity.getRevisions().forEach(revision => revision.setDefaultAppearance(defaultStyle))
     );
   }
+
+  public async findImageAnnotation(
+    filter: Image360AnnotationAssetFilter
+  ): Promise<Image360AnnotationAssetQueryResult[]> {
+    const imageIds = await this._image360DataProvider.getFilesByAssetRef(filter.assetRef);
+    const imageIdSet = new Set<CogniteInternalId>(imageIds);
+
+    const entityAnnotationsPromises = this.image360Entities.map(async entity => {
+      const revisionAndAnnotationPromises = await getEntityAnnotationsForAsset(entity);
+
+      const revisionsAndAnnotations = revisionAndAnnotationPromises.map(async ({ revision, annotation }) => ({
+        annotation,
+        revision,
+        image: entity
+      }));
```

Comment:
Yes, it's the same.. :thinking: I thought this would make it slightly more readable, but it's also adding that ugly last line. I think I'll go with your suggestion

Reply:
Became much simpler :+1: